modules/browsemap/src/main/java/com/sygic/modules/browsemap/viewmodel/BrowseMapFragmentViewModel.kt

```diff
@@ -18,17 +33,71 @@ class BrowseMapFragmentViewModel private constructor(attributesTypedArray: Typed
         attributesTypedArray?.let {
             compassEnabled.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_compassEnabled, false)
             compassHideIfNorthUp.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_compassHideIfNorthUp, false)
+            mapInteractionMode.value =
+                    it.getInt(R.styleable.BrowseMapFragment_sygic_mapInteractionMode, MapInteractionMode.NONE)
             positionLockFabEnabled.value =
                     it.getBoolean(R.styleable.BrowseMapFragment_sygic_positionLockFabEnabled, false)
             zoomControlsEnabled.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_zoomControlsEnabled, false)
             it.recycle()
         }
+
+        mapInteractionMode.value?.let { mode ->
+            when (mode) {
+                MapInteractionMode.FULL, MapInteractionMode.MARKERS_ONLY -> mapInteractionManager.addOnMapClickListener(
+                    this
+                )
+            }
+        }
+    }
+
+    override fun onMapObjectsRequestStarted() {
+        extendedMapDataModel.removeOnClickMapMarker()
+    }
+
+    override fun onMapObjectsReceived(viewObjects: List<ViewObject>) {
+        mapInteractionMode.value?.let { mode ->
+            val firstViewObject = viewObjects.first()
+            when (mode) {
+                MapInteractionMode.NONE -> return
+                MapInteractionMode.MARKERS_ONLY -> {
+                    if (firstViewObject is MapMarker) getPoiData(firstViewObject)
+                }
+                MapInteractionMode.FULL -> {
+                    getPoiData(firstViewObject)
+                    if (firstViewObject is MapMarker) {
+                        return
+                    }
+
+                    extendedMapDataModel.addOnClickMapMarker(OnClickMapMarker(firstViewObject))
+                }
+            }
+        }
+    }
+
+    private fun getPoiData(viewObject: ViewObject) {
+        poiDataManager.getPoiData(viewObject, object : PoiDataManager.Callback() {
+            override fun onDataLoaded(poiData: PoiData) {
+                extendedMapDataModel.notifyPoiDataChanged(poiData)
+            }
+        })
+    }
+
+    override fun onCleared() {
+        mapInteractionManager.removeOnMapClickListener(this)
+        super.onCleared()
```

Comment:
az na vynimky, kedy to ma zmysel, volaj pls super vzdy prvy. Aj ked tu konkretne je super prazdny a nemusel by ani byt volany...

Reply:
Mno, oki. V "create" fázích vždy (nebo skoro vždy) volám super jako první, ale v destroy fázích mi dává smysl si vždy nejprve uklidit po sobě a pak to propustit dále, či?

Reply:
nie je to nejake dolezite, kto uprace prvy, skor je vo vseobecnosti zauzivane, ze najskor nechas vykonat veci parenta, az potom seba. Preto je to pri konstruktoroch forcovane.